airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/base/PythonTestSource.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.io.LineGobbler;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.AirbyteCatalog;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.workers.WorkerUtils;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Extends TestSource such that it can be called using resources pulled from the file system. Will
+ * also add the ability to execute arbitrary scripts in the next version.
+ */
+public class PythonTestSource extends TestSource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PythonTestSource.class);
+  private static final String OUTPUT_FILENAME = "output.json";
+
+  public static String IMAGE_NAME;
+  public static String PYTHON_CONTAINER_NAME;
+
+  @Override
+  protected String getImageName() {
+    return IMAGE_NAME;
+  }
+
+  @Override
+  protected ConnectorSpecification getSpec() throws IOException {
+    return runExecutable(Command.GET_SPEC, ConnectorSpecification.class);
+  }
+
+  @Override
+  protected JsonNode getConfig() throws IOException {
+    return runExecutable(Command.GET_CONFIG);
+  }
+
+  @Override
+  protected AirbyteCatalog getCatalog() throws IOException {
+    return runExecutable(Command.GET_CATALOG, AirbyteCatalog.class);
+  }
+
+  @Override
+  protected void setup(TestDestinationEnv testEnv) throws Exception {
+    runExecutableVoid(Command.SETUP);
+  }
+
+  @Override
+  protected void tearDown(TestDestinationEnv testEnv) throws Exception {
+    runExecutableVoid(Command.TEARDOWN);
+  }
+
+  private enum Command {
+    GET_SPEC,
+    GET_CONFIG,
+    GET_CATALOG,
+    SETUP,
+    TEARDOWN
+  }
+
+  private static <T> T runExecutable(Command cmd, Class<T> klass) throws IOException {
+    return Jsons.object(runExecutable(cmd), klass);
+  }
+
+  private static JsonNode runExecutable(Command cmd) throws IOException {
+    // todo (cgardens) - hack to get around python being python. python is wrapping the json in an extra
```

Comment:
looks like you are serializing the json twice in `standard_test.py`

Reply:
🙌  thanks.